src/main/java/org/myrobotlab/audio/AudioProcessor.java

```diff
@@ -195,21 +199,52 @@ public AudioData play(AudioData data) {
             }
           }
 
-          // BooleanControl
-          // muteControl=(BooleanControl)source.getControl(BooleanControl.Type.MUTE);
-          /*
-           * if (volume == 0) { muteControl.setValue(true); }
-           */
-
-          // the buffer of raw data could be published from here
-          // if a reference of the service is passed in
-
           if (audioFile.isMute()) {
             // NoOp for a mute audioFile.
           } else {
+            AudioFileConfig config = (AudioFileConfig)audioFile.getConfig();
+            if (cnt % config.peakSampleInterval == 0) {
+              float[] samples = new float[buffer.length / 2];

```

Comment:
you shouldn't be creating this array here...  you should just scan over the "sample" and find the peak.